spec/index.bs

```diff
@@ -7213,9 +7218,21 @@ GPUQuerySet includes GPUObjectBase;
         The {{GPUQuerySetDescriptor}} describing this query set.
 
         All optional fields of {{GPUTextureViewDescriptor}} are defined.
+
+    : <dfn>\[[state]]</dfn> of type [=query set state=].
+    ::
+        The current state of the {{GPUQuerySet}}.
 </dl>
 
-### Creation ### {#queryset-creation}
+Each {{GPUQuerySet}} has a current <dfn dfn>query set state</dfn> on the [=Content timeline=]
+which is one of the following:
+
+ - "<dfn dfn for="query set state">available</dfn>" where the {{GPUQuerySet}} is
+     available for GPU operations on its content.
+ - "<dfn dfn for="query set state">destroyed</dfn>" where the {{GPUQuerySet}} is
+     no longer available for any operations except {{GPUQuerySet/destroy}}.
```

Comment:
Perhaps this could just be a simple `boolean [[destroyed]]`. Or will other states be needed here later?

Reply:
I don't think it would be weird, but I thought about this further in relation to the other destroyable objects (buffer and texture). I think let's keep the states for consistently. Each object can be:

- boolean: valid or invalid
- state enum: destroyed, or in some other state (available, "mapping pending", etc.)